lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexDriver.java

```diff
@@ -0,0 +1,180 @@
+import org.apache.reef.driver.evaluator.*;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Configurations;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.annotations.Unit;
+import org.apache.reef.vortex.api.VortexThreadPool;
+import org.apache.reef.vortex.api.VortexUserCode;
+import org.apache.reef.vortex.common.ReportTaskletResult;
+import org.apache.reef.vortex.driver.tasklet.DefaultTaskletExecutor;
+import org.apache.reef.vortex.evaluator.ExecutorConfiguration;
+import org.apache.reef.vortex.evaluator.LocalExecutor;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.StartTime;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * REEF Driver for Vortex.
+ */
+@Unit
+public final class VortexDriver {
+  private static final Logger LOG = Logger.getLogger(VortexDriver.class.getName());
+  private final EvaluatorRequestor evaluatorRequestor;
+  private final VortexManager vortexManager;
+  private final CommandSender commandSender;
+
+  private final int evalMem;
+  private final int evalNum;
+  private final int evalCores;
+
+  private final VortexThreadPool vortexThreadPool;
+  private final ExecutorService vortexUserThread = Executors.newSingleThreadExecutor();
+  private final VortexUserCode vortexUserCode;
+
+  @Inject
+  public VortexDriver(final EvaluatorRequestor evaluatorRequestor,
+                      final CommandSender commandSender,
+                      final VortexManager vortexManager,
+                      final VortexUserCode vortexUserCode,
+                      final VortexThreadPool vortexThreadPool,
+                      @Parameter(VortexConfiguration.ExecutorMem.class) final int evalMem,
+                      @Parameter(VortexConfiguration.ExecutorNum.class) final int evalNum,
+                      @Parameter(VortexConfiguration.ExecutorCores.class) final int evalCores) {
+    this.evaluatorRequestor = evaluatorRequestor;
+    this.vortexManager = vortexManager;
+    this.commandSender = commandSender;
+    this.vortexUserCode = vortexUserCode;
+    this.vortexThreadPool = vortexThreadPool;
+    this.evalMem = evalMem;
+    this.evalNum = evalNum;
+    this.evalCores = evalCores;
+  }
+
+  /**
+   * Driver started.
+   */
+  public final class StartHandler implements EventHandler<StartTime> {
+    @Override
+    public void onNext(final StartTime startTime) {
+      // Initial Evaluator Request
+      evaluatorRequestor.submit(EvaluatorRequest.newBuilder()
+          .setNumber(evalNum)
+          .setMemory(evalMem)
+          .setNumberOfCores(evalCores)
+          .build());
+
+      // Run the user code as a separate thread
+      // todo: handle user code exceptions
+      vortexUserThread.execute(new Runnable() {
```

Comment:
Yes, now we call
```java
vortexStartEStage.onNext(vortexStart);
```
in the constructor. The code is moved out of `VortexDriver` to there.